code/part-one/signing.js

```diff
@@ -23,12 +23,17 @@ const getPublicKey = privateKey => {
   return secp256k1.publicKeyCreate(toBytes(privateKey)).toString('hex');
 };
 
-const sign = (message, privateKey) => {
-
+const sign = (privateKey, message) => {
+  const { signature } = secp256k1.sign(sha256(message), toBytes(privateKey));
```

Comment:
Should we order the arguments on ln 26 in the same way we pass them into `secp256k1.sign(...)` on ln 27 (aka message, then privateKey)?

Reply:
Open to debate. Coming from a functional perspective, the order `secp256k1-node` uses isn't great. If you are going to curry or partially apply your function, you are going to want to do the private key first so you can sign multiple messages with the same partially applied function. From that perspective, I did every function with key-first arguments.

That said, it could certainly be a point of confusion for students. I liked the consistent key-first approach enough that I felt it overcame any momentary confusion, but I could be talked out of it.

Reply:
I think it'd be cool to add a comment about this exact reasoning re: why you ordered the arguments in that way!